ProfilerGUI/Source/Configurator/TargetAppModel.cs

```diff
@@ -120,21 +121,51 @@ private void UpdateExecutableMachineType()
                 return;
             }
 
-            MachineTypeUtils.MachineType typeOfExecutable = MachineTypeUtils.GetExecutableType(ApplicationPath);
-            if (typeOfExecutable == MachineTypeUtils.MachineType.Unknown)
-            {
-                logger.Warn("Could not determine if target app is 32 or 64 bit, please set manually");
-                return;
-            }
-
-            if (typeOfExecutable == MachineTypeUtils.MachineType.I386)
-            {
-                ApplicationType = EApplicationType.Type32Bit;
-            }
-            else
-            {
-                ApplicationType = EApplicationType.Type64Bit;
+            if (IsDotNETCoreApplication(ApplicationPath))
+            {
+                ApplicationType = EApplicationType.TypeCORE;
+            }
+            else
+            {
+                MachineTypeUtils.MachineType typeOfExecutable = MachineTypeUtils.GetExecutableType(ApplicationPath);
+                if (typeOfExecutable == MachineTypeUtils.MachineType.Unknown)
+                {
+                    logger.Warn("Could not determine if target app is 32 or 64 bit, please set manually");
+                    return;
+                }
+
+                if (typeOfExecutable == MachineTypeUtils.MachineType.I386)
+                {
+                    ApplicationType = EApplicationType.Type32Bit;
+                }
+                else
+                {
+                    ApplicationType = EApplicationType.Type64Bit;
+                }
             }
+        }
+        
+        private static bool IsDotNETCoreApplication(string applicationPath)
+        {
+            try
+            {
+                // If the application is .NET, we can load it as an assembly.
+                var assembly = Assembly.LoadFile(applicationPath);

```

Comment:
Maybe it would be better to use Mono.Cecil (http://cecil.pe/) instead of `Assembly.Load` here? Afaik the latter will also load dependent assemblies in RAM / current VM

Reply:
I switched to using [`Assembly.ReflectionOnlyLoadFrom`](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.assembly.reflectiononlyloadfrom?view=netframework-4.7.2), since it saves us the additional dependency, avoids that dependencies are loaded, and prevents execution of any code from the assembly.

Reply:
Good catch, wasn't aware of this method at all. That'S the point where reviews make perfectly sense :)